### PR TITLE
set_dependencies: fix KeyError with missing "call" result

### PR DESCRIPTION
**Pull request by blueyed**
"call" might not be there when quitting during setup already.

Fixes:

    INTERNALERROR>   File "…/Vcs/pytest-testmon/testmon/pytest_testmon.py", line 220, in pytest_runtest_protocol
    INTERNALERROR>     self.reports[item.nodeid])
    INTERNALERROR>   File "…/Vcs/pytest-testmon/testmon/testmon_core.py", line 311, in stop_and_save
    INTERNALERROR>     testmon_data.set_dependencies(nodeid, testmon_data.get_nodedata(nodeid, self.cov.get_data(), rootdir), result)
    INTERNALERROR>   File "…/Vcs/pytest-testmon/testmon/testmon_core.py", line 240, in set_dependencies
    INTERNALERROR>     json.dumps([result['setup'], result['call'], result['teardown']]) if outcome else '[]',
    INTERNALERROR> KeyError: 'call'

### CODE REVIEW

File: testmon/testmon_core.py [hunk 1]
```diff
@@ -230,15 +230,23 @@ def get_nodedata(self, nodeid, coverage_data, rootdir):
...
     def set_dependencies(self, nodeid, nodedata, result=None):
         with self.connection as con:
-            outcome = bool([True for r in result.values() if r.get('outcome') == u'failed'])
+            failed = any(r.get('outcome') == u'failed'
+                         for r in result.values())
+            if failed:
+                results_json = '[]'
+            else:
+                results = []
+                for k in ['setup', 'call', 'teardown']:
```
Comment (blueyed) [inline testmon/testmon_core.py]: I assume keeping the order is important here?!

**Comment by blueyed** [comment]
Sounds good.

**Comment by blueyed** [comment]
But then the order might not be retained, which might be important because `setup` should get reported before `teardown`, no?

**Comment by tarpas** [comment]
I don't know, but maybe this is a solution: https://stackoverflow.com/questions/6921699/can-i-get-json-to-load-into-an-ordereddict

**Comment by blueyed** [comment]
I think keeping it as a list is simpler.

**Comment by tarpas** [comment]
replaced with other pull request, thanks for pointing out the bug.

**Pull request closed without merging**